spec/migrations/20200331150436_rename_foreman_features_spec.rb

```diff
@@ -0,0 +1,68 @@
+require_migration
+
+describe RenameForemanFeatures do
+  let(:user_role_id) { anonymous_class_with_id_regions.id_in_region(1, anonymous_class_with_id_regions.my_region_number) }
+  let(:feature_stub) { migration_stub :MiqProductFeature }
+  let(:roles_feature_stub) { migration_stub :MiqRolesFeature }
+  let(:user_stub) { migration_stub :User }
+
+  migration_context :up do
+    it 'renames the features' do
+      accord_feature = feature_stub.create!(:identifier => 'providers_accord')
+      view_feature = feature_stub.create!(:identifier => 'provider_foreman_view')
+
+      migrate
+
+      expect(accord_feature.reload.identifier).to eq('ems_configuration')
+      expect(view_feature.reload.identifier).to eq('ems_configuration_view')
+    end
+
+    it 'appends the extra features for roles' do
+      %w[ems_configuration configured_system configuration_profile].each do |feature|
+        feature_stub.create!(:identifier => feature)
+      end
+
+      explorer_feature = feature_stub.create!(:identifier => 'provider_foreman_explorer')
+      roles_feature_stub.create!(:miq_product_feature_id => explorer_feature.id, :miq_user_role_id => user_role_id)
+
+      expect(roles_feature_stub.where(:miq_user_role_id => user_role_id).count).to eq(1)
+
+      migrate
+
+      expect(roles_feature_stub.where(:miq_user_role_id => user_role_id).count).to eq(4)
+    end
+
+    it 'replaces user starting page if foreman' do
+      feature_stub.create!(:identifier => 'provider_foreman_explorer')
+      user = user_stub.create!(:settings => {:display => {:startpage => 'provider_foreman/explorer'}})
```

Comment:
Does there need to be a test for a user that doesn't have the settings at all, or perhaps doesn't have the :settings => :display section?  Is that even a possibility?

Reply:
User always have default settings https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/application_controller.rb#L100, default (startpage setting)[https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/controllers/application_controller.rb#L193]

Reply:
@h-kataria Those are defined as defaults in the controller and so are not available at migration time.  My question is whether those values are always present in the column.  That is, can the column value be nil or an empty hash, thus causing this code to blow up?

Reply:
@Fryguy I am doing the check for the settings in the safest way possible:
```ruby
if user.settings&.dig(:display, :startpage) == 'provider_foreman/explorer'
```